cmd/lotus/rpc.go

```diff
@@ -44,3 +55,30 @@ func serveRPC(a api.FullNode, stop node.StopFunc, addr string) error {
 
 	return srv.ListenAndServe()
 }
+
+func handleImport(a api.FullNode) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "PUT" {
+			w.WriteHeader(404)
+			return
+		}
+		if !api.HasPerm(r.Context(), api.PermWrite) {
+			w.WriteHeader(401)
+			json.NewEncoder(w).Encode(struct{ Error string }{"unauthorized: missing write permission"})
+			return
+		}
+
+		c, err := a.(*impl.FullNodeAPI).ClientImportLocal(r.Context(), r.Body)
```

Comment:
why assert here?

Reply:
`ClientImportLocal` is not exposed on the api interface / jsonrpc (because `io.Reader`)

Reply:
IMO, it would be better to do that cast in L37, instead of here.